### DISCUSSION TITLE
Set keybinding to run several command successively

### DISCUSSION THREAD

**Discussion by yuandi42** [qutebrowser/qutebrowser#7062] [Q&A] [answered]
I've written a dmenu script to disable/enable grease moneky scripts, and set two keybinding to spawn that script and `greasemonkey-reload` respectively. Now I wonder if there is a way to set just one keybings to spawn those two command in order for convenience.

**Comment by mschilli87** [comment]
I'm not sure I understood your use case completely, but in principle you should be able to join as many commands as you want separated by a semicolon (`;`) for any key binding.

**Reply by yuandi42** [reply]
Excuse me, what does "separated by a semicolon" mean? I typed below in my config:
```python
config.bind('gs','spawn --userscript dmenu_gms ; greasemonkey-reload' , mode = 'normal')
```
It only spawn the script and doesn't run `greasemonkey-reload`.
Then I tried (in a stupid way)
```python
config.bind('gs','spawn --userscript dmenu_gms'; ' greasemonkey-reload' , mode = 'normal')
```
And I got a error message when the browser was opened, not surprised though.
I've never been good at programming and for certain I made a stupid mistake. What should I do?

**Reply by mschilli87** [reply]
Sorry, I might have mixed things up. IIRC, you need to double the semicolon.

Could you try the following:

```python
config.bind('gs', 'spawn --userscript dmenu_gms ;; greasemonkey-reload', mode = 'normal')
```

?

**Reply by yuandi42** [reply]
It works anyway, but not in the way I want. 
Actually, pressing `gs` runs two command at the same time, which means greasemonkey scripts reload when I'm still selecting which scripts to enable/disable in dmenu. What I need is, run `greasemonkey-reload` **after** I select a greasemonkey script , i.e. finally exit the userscript. Is there a way to do that?

**Reply by mschilli87** [reply]
I guess the issue is tha `spawn` doesn't block until `dmenu` exits. To check that, you could add a `sleep` command in between that's long enough to do your selection. If that is the case, then you'd need some way to communicate back from your userscript. I don't think there is a nice way to do so. There are many nasty hacks you could do but I think the cleanest way would be to 'simply' trigger `greasemonkey-reload` from within your `dmenu_gms` script. Have you tried a simple

```sh
qutebrowser ':greasemonkey-releoad'
```

? Since the command is launched by qutebrowser there will be a session to run the command in. Not sure how this would behave with several qutebowser sessions open at the same time though.

**Reply by The-Compiler** [reply]
Userscripts have access to `$QUTE_FIFO` precisely to send commands to qutebrowser.

**Comment by yuandi42** [comment, marked as answer]
So just adding `echo "greasemonkey-reload" > "$QUTE_FIFO"` into my script solved the problems. Thx for Ur patience!
